pkg/controllers/util/federatedinformer.go

```diff
@@ -618,3 +623,54 @@ func (fs *federatedStoreImpl) ClustersSynced(clusters []*fedcorev1a1.FederatedCl
 	}
 	return true
 }
+
+func (fs *federatedStoreImpl) ClusterSynced(clusterName string) bool {
+	fs.federatedInformer.Lock()
+	defer fs.federatedInformer.Unlock()
+
+	if targetInformer, found := fs.federatedInformer.targetInformers[clusterName]; found {
+		return targetInformer.controller.HasSynced()
+	}
+
+	return false
+}
+
+// GetClusterObject is a helper function to get a cluster object. GetClusterObject first attempts to get the object from
+// the federated informer with the given key. However, if the cache for the cluster is not synced, it will send a GET
+// request to the cluster's apiserver to retrieve the object directly.
+func GetClusterObject(
+	ctx context.Context,
+	informer FederatedInformer,
+	clusterName string,
+	key string,
+	qualifedName common.QualifiedName,
+	apiResource metav1.APIResource,
+) (*unstructured.Unstructured, bool, error) {
+	if informer.GetTargetStore().ClusterSynced(clusterName) {
+		clusterObj, exists, err := informer.GetTargetStore().GetByKey(clusterName, key)
+		if clusterObj != nil {
```

Comment:
Any chance `clusterObj` will be non-nil garbage when `!exists || err != nil`?

Reply:
Good point. We should perform an error check here.